ring_theory/localization.lean

```diff
@@ -219,14 +333,43 @@ by refine
   inv_mul_cancel := quotient.ind _,
   has_decidable_eq := localization.decidable_eq β,
   inv_zero := dif_pos rfl,
-  .. localization.comm_ring β _ };
+  .. localization.comm_ring };
 { intros x hnx,
   rcases x with ⟨x, z, hz⟩,
   have : x ≠ 0,
     from assume hx, hnx (quotient.sound $ r_of_eq $ by simp [hx]),
-  simp [has_inv.inv, inv_aux, inv_aux._match_1, this],
-  exact (quotient.sound $ r_of_eq $ by simp; ring) }
+  simp only [has_inv.inv, inv_aux, quotient.lift_beta, dif_neg this],
+  exact (quotient.sound $ r_of_eq $ by simp [mul_comm]) }
+
+@[simp] lemma div_eq_div {r s} : (div r s : quotient_ring β) =
+  @has_div.div (quotient_ring β)
+    (@division_ring_has_div (quotient_ring β)
+      (@field.to_division_ring (quotient_ring β) (@discrete_field.to_field (quotient_ring β) (quotient_ring.field.of_integral_domain β)))
+      (@field.to_division_ring (quotient_ring β) (@discrete_field.to_field (quotient_ring β) (quotient_ring.field.of_integral_domain β))))
```

Comment:
Uuugh, what went wrong? That's an awful amount of `@`s.

Reply:
If it requires this many `@`s to state the theorem something is wrong with the interface.

Reply:
It seems to me that this is the only non-trivial issue with this PR.

Reply:
done